### DISCUSSION TITLE
auto generated fields should be readonly by default

### DISCUSSION THREAD

**Discussion by sbts** [ledgersmb/LedgerSMB#5857] [Ideas]
Examples of these are the Order Number, Invoice Number, Control-Code etc fields.
Basically anything that the software would "normally" autogenerate and supply a value for.

Where it may make sense to allow that autogenerated value to be overridden, having a "pen" glyph to the right of the field that "unlocks" editing would be one option.

Either way, these fields should be visually distinct with a different background/foreground and/or a different border

**Comment by ehuelsmann** [comment]
What UI do you propose we use to model this?

**Comment by sbts** [comment]
There are many that work in similar ways to what I've proposed.
I don't have anything specific to point at, but parts of Webmin, and SugarCRM come to mind

**Comment by ehuelsmann** [comment]
I actually have 2 problems preventing implementation any time soon:

1. I miss the function in dojo widget behaviour to mark the field editable (the traditional pencil)
2. We'd have to go through every screen identifying which fields are auto generated
